ctpolicy/chromepolicy.go

```diff
@@ -0,0 +1,130 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package ctpolicy contains structs describing CT policy requirements and
+// corresponding logic.
+package ctpolicy
+
+import (
+	"github.com/google/certificate-transparency-go/loglist"
+	"github.com/google/certificate-transparency-go/x509"
+)
+
+// googGroupBuilder is used to build log-group covering Google-operated logs.
+type googGroupBuilder struct {
+}
+
+func (g googGroupBuilder) GetName() string {
+	return "Google-operated"
+}
+func (g googGroupBuilder) GetDescription() string {
+	return "All logs that are operated by Google."
+}
+func (g googGroupBuilder) IsMember(log *loglist.Log) bool {
+	return log.ByGoogle()
+}
+func (g googGroupBuilder) IsBase() bool {
+	return false
+}
+
+// nonGoogGroupBuilder is used to build log-group covering not-Google-operated logs.
+type nonGoogGroupBuilder struct {
+}
+
+func (g nonGoogGroupBuilder) GetName() string {
+	return "Non-Google-operated"
+}
+func (g nonGoogGroupBuilder) GetDescription() string {
+	return "All logs that are NOT operated by Google."
+}
+func (g nonGoogGroupBuilder) IsMember(log *loglist.Log) bool {
+	return !log.ByGoogle()
+}
+func (g nonGoogGroupBuilder) IsBase() bool {
+	return false
+}
+
+// ChromeCTPolicy incapsulates Chrome Certificate Policy. See https://github.com/chromium/ct-policy/blob/master/ct_policy.md#qualifying-certificate
+// Instantiated for a particular LogList.
+type ChromeCTPolicy struct {
+	googGroup    LogGroup
+	nonGoogGroup LogGroup
+	baseGroup    LogGroup
+	ll           *loglist.LogList
+}
+
+// Groups returns map of log-groups policy operates onto.
+func (chromeP *ChromeCTPolicy) Groups() map[string]*LogGroup {
+	m := make(map[string]*LogGroup)
+	m[chromeP.googGroup.name] = &chromeP.googGroup
+	m[chromeP.nonGoogGroup.name] = &chromeP.nonGoogGroup
+	m[chromeP.baseGroup.name] = &chromeP.baseGroup
+	return m
+}
+
+// GroupMinInclusions returns map from log-group names to minimal required number of logs within that group.
+// Numbers come from https://github.com/chromium/ct-policy/blob/master/ct_policy.md#qualifying-certificate
```

Comment:
Need some reminders scattered around this package to indicate the policy stuff is for embedded SCTs in particular.

Reply:
Is there a comment somewhere about this?